src/librustc/mir/mod.rs

```diff
@@ -395,6 +411,16 @@ pub struct LocalDecl<'tcx> {
     /// True if this corresponds to a user-declared local variable.
     pub is_user_variable: bool,
 
+    /// True if this is an internal local.
```

Comment:
I would have preferred a comment that "this is sound because the drop flags are fully algebraic, and therefore don't affect the OIBIT or outlives properties of the generator". It's nice to state your invariants.